---
title: "LeafSectionExperiments"
author: "Devon Birdseye"
date: '2023-05-17'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r packages}
library(factoextra)
library(dplyr)
library(ggbiplot)
library(reshape2)
library(ggridges)
```
```{r mgdb}
#Downloaded from MaizeGDB 2020/05/18, 07:11pm
mgdb <- read.csv("mgdb.csv", stringsAsFactors = F, header = F)
colnames(mgdb) <- mgdb[5,]
mgdb <- mgdb[-c(1:5),]
```
```{r Genes Of Interest}
GOIs <- readRDS("GOIs")
GOIs$PhANGs <- c(GOIs$PhANGs, "Zm00001d000279")
```
#Experiment 1
##import data
```{r}
tmt <- read.csv("B73x5-Mo17x5-BMx5 leaf sections TMT15 111521.csv", stringsAsFactors = F)

tmt <- tmt[-grep("only", tmt$Accession),]
tmt[tmt==0] <- NA
tmt <- na.omit(tmt)

tmt$Gene <- substr(tmt$Accession, start = 1, stop = 14)
tmt <- merge.data.frame(tmt, mgdb[,c("v4_gene_model","Entrez")], by.x="Gene", by.y = "v4_gene_model", all.x = T, all.y = F)
```

##Changing normalization method (?)
```{r}
normF <- c(1,1.18006729476876,1.37937113622264,1.46860614798218,1.33153251665837,1.19116920988042,1.16699836139188,1.53996200344565,1.58522393410753,1.47254695173594,0.851821568490102,1.22377090843489,1.27086418901632,1.45697842359028,1.29980500363454)
```
```{r}
tmt$B73.1 <- tmt$B73.1*normF[1]
tmt$B73.2 <- tmt$B73.2*normF[2]
tmt$B73.3 <- tmt$B73.3*normF[3]
tmt$B73.4 <- tmt$B73.4*normF[4]
tmt$B73.5 <- tmt$B73.5*normF[5]

tmt$Mo17.1 <- tmt$Mo17.1*normF[6]
tmt$Mo17.2 <- tmt$Mo17.2*normF[7]
tmt$Mo17.3 <- tmt$Mo17.3*normF[8]
tmt$Mo17.4 <- tmt$Mo17.4*normF[9]
tmt$Mo17.5 <- tmt$Mo17.5*normF[10]

tmt$BM.1 <- tmt$BM.1*normF[11]
tmt$BM.2 <- tmt$BM.2*normF[12]
tmt$BM.3 <- tmt$BM.3*normF[13]
tmt$BM.4 <- tmt$BM.4*normF[14]
tmt$BM.5 <- tmt$BM.5*normF[15]
```
```{r}
tmt$B73.1 <- tmt$B73.1/sum(tmt$B73.1)
tmt$B73.2 <- tmt$B73.2/sum(tmt$B73.2)
tmt$B73.3 <- tmt$B73.3/sum(tmt$B73.3)
tmt$B73.4 <- tmt$B73.4/sum(tmt$B73.4)
tmt$B73.5 <- tmt$B73.5/sum(tmt$B73.5)

tmt$Mo17.1 <- tmt$Mo17.1/sum(tmt$Mo17.1)
tmt$Mo17.2 <- tmt$Mo17.2/sum(tmt$Mo17.2)
tmt$Mo17.3 <- tmt$Mo17.3/sum(tmt$Mo17.3)
tmt$Mo17.4 <- tmt$Mo17.4/sum(tmt$Mo17.4)
tmt$Mo17.5 <- tmt$Mo17.5/sum(tmt$Mo17.5)

tmt$BM.1 <- tmt$BM.1/sum(tmt$BM.1)
tmt$BM.2 <- tmt$BM.2/sum(tmt$BM.2)
tmt$BM.3 <- tmt$BM.3/sum(tmt$BM.3)
tmt$BM.4 <- tmt$BM.4/sum(tmt$BM.4)
tmt$BM.5 <- tmt$BM.5/sum(tmt$BM.5)
```

##calculate hyb/MP ratios
```{r}
tmt$MP.1 <- rowMeans(tmt[,c("B73.1", "Mo17.1")])
tmt$MP.2 <- rowMeans(tmt[,c("B73.2", "Mo17.2")])
tmt$MP.3 <- rowMeans(tmt[,c("B73.3", "Mo17.3")])
tmt$MP.4 <- rowMeans(tmt[,c("B73.4", "Mo17.4")])
tmt$MP.5 <- rowMeans(tmt[,c("B73.5", "Mo17.5")])

tmt$l2.BM.1.MP.1 <- log2(tmt$BM.1/tmt$MP.1)
tmt$l2.BM.2.MP.2 <- log2(tmt$BM.2/tmt$MP.2)
tmt$l2.BM.3.MP.3 <- log2(tmt$BM.3/tmt$MP.3)
tmt$l2.BM.4.MP.4 <- log2(tmt$BM.4/tmt$MP.4)
tmt$l2.BM.5.MP.5 <- log2(tmt$BM.5/tmt$MP.5)
```
##calculate B73/Mo17 ratios
```{r}
tmt$l2.B73.1.Mo17.1 <- log2(tmt$B73.1/tmt$Mo17.1)
tmt$l2.B73.2.Mo17.2 <- log2(tmt$B73.2/tmt$Mo17.2)
tmt$l2.B73.3.Mo17.3 <- log2(tmt$B73.3/tmt$Mo17.3)
tmt$l2.B73.4.Mo17.4 <- log2(tmt$B73.4/tmt$Mo17.4)
tmt$l2.B73.5.Mo17.5 <- log2(tmt$B73.5/tmt$Mo17.5)
```

```{r}
write.csv(na.omit(tmt[!tmt$Entrez=="",c("Gene","Accession","Entrez")]), row.names = F, "tmt.EnrichmentBackground.csv")
```
##PCA
```{r}
tmt.pca <- t(tmt[,2:16])
colnames(tmt.pca) <- tmt$Accession
tmt.pca <- scale(tmt.pca, scale = T, center = F)
tmt.pca <- prcomp(tmt.pca)
```
```{r}
fviz_eig(tmt.pca, addlabels = TRUE)
#ggsave("LeafSectionsScree.png", width = 7, height = 5)
```
```{r}
tmt.gt <- c(rep("B73",5), rep("Mo17",5), rep("BxM",5))
```
```{r}
ggbiplot(tmt.pca, var.axes = F, labels = colnames(tmt[,2:16]), groups = tmt.gt)+
  theme_bw()+
  theme(legend.position = "none")
#ggsave("LeafSectionsPCA.png", width = 4, height = 4)
```

```{r}
tmt.pca.var <- get_pca_var(tmt.pca)
tmt.pca.var.contrib <- as.data.frame(tmt.pca.var$contrib)


tmt.pca.var.contrib$Accession <- rownames(tmt.pca.var.contrib)
tmt.pca.var.contrib$Gene <- substr(tmt.pca.var.contrib$Accession, start = 1, stop = 14)
tmt.pca.var.contrib$Category <- "Other"
tmt.pca.var.contrib$Category[tmt.pca.var.contrib$Gene %in% GOIs$plastid] <- "Plastid-localized"
tmt.pca.var.contrib$Category[tmt.pca.var.contrib$Gene %in% GOIs$PhANGs] <- "PhANGs"
tmt.pca.var.contrib$Category[tmt.pca.var.contrib$Gene %in% GOIs$PhAPGs] <- "PhAPGs"
tmt.pca.var.contrib$Category[tmt.pca.var.contrib$Gene %in% GOIs$NE.PtRibo] <- "NE plastid ribosome"
tmt.pca.var.contrib$Category[tmt.pca.var.contrib$Gene %in% GOIs$PE.ribo] <- "PE plastid ribosome"
tmt.pca.var.contrib$Category[tmt.pca.var.contrib$Gene %in% GOIs$EtBio] <- "Ethylene biosynthesis"

tmt.pca.var.contrib.Dim1 <- merge.data.frame(tmt.pca.var.contrib[,c("Accession", "Category", "Dim.1")], tmt[,c("Accession", "Gene", "Entrez", "Protein")], by.x = "Accession", by.y = "Accession", all.x = T, all.y = F)
tmt.pca.var.contrib.Dim1 <- tmt.pca.var.contrib.Dim1[order(-tmt.pca.var.contrib.Dim1$Dim.1),]
write.csv(tmt.pca.var.contrib.Dim1, row.names = F, "tmt.pca.var.contrib.Dim1.csv")

tmt.pca.var.contrib.Dim2 <- merge.data.frame(tmt.pca.var.contrib[,c("Accession", "Category", "Dim.2")], tmt[,c("Accession", "Gene", "Entrez", "Protein")], by.x = "Accession", by.y = "Accession", all.x = T, all.y = F)
tmt.pca.var.contrib.Dim2 <- tmt.pca.var.contrib.Dim2[order(-tmt.pca.var.contrib.Dim2$Dim.2),]
write.csv(tmt.pca.var.contrib.Dim2, row.names = F, "tmt.pca.var.contrib.Dim2.csv")
```


```{r}
tmt.pca.var.contrib$Category <- factor(tmt.pca.var.contrib$Category, levels = c("Other", "Plastid-localized", "Ethylene biosynthesis","PE plastid ribosome", "NE plastid ribosome", "PhAPGs","PhANGs"))
ggplot(tmt.pca.var.contrib, mapping=aes(y=Category, x=Dim.1, fill = stat(x)))+
  geom_density_ridges_gradient(size=1) +
  scale_fill_viridis_c(option = "C")+
  theme_ridges() +
  theme(axis.text =element_text(size=8), axis.title = element_text(size=8), legend.position = "none", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=8))+
  xlim(0,max(tmt.pca.var.contrib$Dim.1))+
  coord_cartesian(expand = F)
```
```{r}
ggplot(tmt.pca.var.contrib, mapping=aes(y=Category, x=Dim.2, fill = stat(x)))+
  geom_density_ridges_gradient(alpha=0.7) +
  scale_fill_viridis_c(option = "B")+
  theme_ridges() +
  theme(axis.text =element_text(size=8), axis.title = element_text(size=8), legend.position = "none", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=8))
```
##Expression Heterosis Density Curves
```{r}
tmt.BM.MP <- tmt[,c("Gene","l2.BM.1.MP.1", "l2.BM.2.MP.2","l2.BM.3.MP.3","l2.BM.4.MP.4","l2.BM.5.MP.5")]
colnames(tmt.BM.MP) <- c("Gene","Section1", "Section2","Section3","Section4","Section5")
tmt.BM.MP.m <- melt(tmt.BM.MP)

tmt.BM.MP.m$Category <- "Other"

tmt.BM.MP.m$Category[tmt.BM.MP.m$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs)] <- "PhANGs/PhAPGs"

tmt.BM.MP.m$Category[tmt.BM.MP.m$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo)] <- "Plastid ribosome"


tmt.BM.MP.m$Category[tmt.BM.MP.m$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"


colors.p2 <- c("Other"="grey50",  "Plastid ribosome"="#6A00EA",  "PhANGs/PhAPGs"="#008275", "Cytosolic ribosome"="#B50000")

tmt.BM.MP.m$value[tmt.BM.MP.m$value <= -2] <- -2
tmt.BM.MP.m$value[tmt.BM.MP.m$value >= 2] <- 2

ggplot(tmt.BM.MP.m, mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p2, name="")+
  theme_ridges() +
  xlim(-2,2)+
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(BM/MP)")+
  guides(col=guide_legend(nrow = 2))+
  coord_cartesian(clip = "off", expand = F)
ggsave("tmt.BM.MP.png", width = 4, height = 5)
```

```{r}
tmt.BM.MP <- tmt[,c("Gene","l2.BM.2.MP.2","l2.BM.3.MP.3","l2.BM.4.MP.4","l2.BM.5.MP.5")]
colnames(tmt.BM.MP) <- c("Gene","Section1", "Section2","Section3","Section4")
tmt.BM.MP.m <- melt(tmt.BM.MP)

tmt.BM.MP.m$Category <- "Other"

tmt.BM.MP.m$Category[tmt.BM.MP.m$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs)] <- "PhANGs/PhAPGs"

tmt.BM.MP.m$Category[tmt.BM.MP.m$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo)] <- "Plastid ribosome"


tmt.BM.MP.m$Category[tmt.BM.MP.m$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"


colors.p2 <- c("Other"="grey50",  "Plastid ribosome"="#6A00EA",  "PhANGs/PhAPGs"="#008275", "Cytosolic ribosome"="#B50000")

tmt.BM.MP.m$value[tmt.BM.MP.m$value <= -2] <- -2
tmt.BM.MP.m$value[tmt.BM.MP.m$value >= 2] <- 2

ggplot(tmt.BM.MP.m, mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p2, name="")+
  theme_ridges() +
  xlim(-2,2)+
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(BM/MP)")+
  guides(col=guide_legend(nrow = 2))+
  coord_cartesian(clip = "off", expand = F)
ggsave("tmt.BM.MP.4sectionsOnly.png", width = 4, height = 5)
```

```{r}
tmt.BM.MP <- tmt[,c("Accession","Gene","l2.BM.1.MP.1", "l2.BM.2.MP.2","l2.BM.3.MP.3","l2.BM.4.MP.4","l2.BM.5.MP.5")]
colnames(tmt.BM.MP) <- c("Accession","Gene","Section1","Section2","Section3","Section4","Section5")
tmt.BM.MP.m <- melt(tmt.BM.MP)

tmt.BM.MP.m$Category <- "Other"
tmt.BM.MP.m$Order <- 1

tmt.BM.MP.m$Category[tmt.BM.MP.m$Gene %in% GOIs$plastid] <- "Plastid-localized"
tmt.BM.MP.m$Order[tmt.BM.MP.m$Gene %in% GOIs$plastid] <- 2

tmt.BM.MP.m$Category[tmt.BM.MP.m$Gene %in% GOIs$PhANGs] <- "PhANGs"
tmt.BM.MP.m$Order[tmt.BM.MP.m$Gene %in% GOIs$PhANGs] <- 3

tmt.BM.MP.m$Category[tmt.BM.MP.m$Gene %in% GOIs$PhAPGs] <- "PhAPGs"
tmt.BM.MP.m$Order[tmt.BM.MP.m$Gene %in% GOIs$PhAPGs] <- 4

tmt.BM.MP.m$Category[tmt.BM.MP.m$Gene %in% GOIs$NE.PtRibo] <- "NE plastid ribosome"
tmt.BM.MP.m$Order[tmt.BM.MP.m$Gene %in% GOIs$NE.PtRibo] <- 5

tmt.BM.MP.m$Category[tmt.BM.MP.m$Gene %in% GOIs$PE.ribo] <- "PE plastid ribosome"
tmt.BM.MP.m$Order[tmt.BM.MP.m$Gene %in% GOIs$PE.ribo] <- 6

colors.p <- c("Other"="#656565", "Plastid-localized"="black", "PE plastid ribosome"="#6A00EA", "NE plastid ribosome"="#E5A3FF", "PhAPGs"="#008275", "PhANGs"="#81E401")

tmt.BM.MP.m$value[tmt.BM.MP.m$value > 2] <- 2
tmt.BM.MP.m$value[tmt.BM.MP.m$value < -2] <- -2

ggplot(tmt.BM.MP.m[tmt.BM.MP.m$Category %in% c("Plastid-localized","PhANGs","PhAPGs"),], mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p[c("Plastid-localized","PhANGs","PhAPGs")], name="")+
  theme_ridges() +
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(BM/MP)")+
  xlim(-2,2)
ggsave("LeafSections_PhAGs_BM_MP.jpg", width = 5, height = 6)

ggplot(tmt.BM.MP.m[tmt.BM.MP.m$Category %in% c("Plastid-localized","NE plastid ribosome","PE plastid ribosome"),], mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p[c("Plastid-localized","NE plastid ribosome","PE plastid ribosome")], name="")+
  theme_ridges() +
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(BM/MP)")+
  xlim(-2,2)
ggsave("LeafSections_pRibo_BM_MP.jpg", width = 5, height = 6)
```
```{r}
tmt.BM.MP <- tmt[,c("Accession","Gene","l2.BM.1.MP.1", "l2.BM.2.MP.2","l2.BM.3.MP.3","l2.BM.4.MP.4","l2.BM.5.MP.5")]
colnames(tmt.BM.MP) <- c("Accession","Gene","Section1","Section2","Section3","Section4","Section5")
tmt.BM.MP.m <- melt(tmt.BM.MP)

tmt.BM.MP.m$Category <- "Other"
tmt.BM.MP.m$Order <- 1

tmt.BM.MP.m$Category[tmt.BM.MP.m$Gene %in% GOIs$EtBio] <- "Ethylene biosynthesis"
tmt.BM.MP.m$Order[tmt.BM.MP.m$Gene %in% GOIs$EtBio] <-2

tmt.BM.MP.m$Category[tmt.BM.MP.m$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"
tmt.BM.MP.m$Order[tmt.BM.MP.m$Gene %in% GOIs$CytoRibo] <-3

colors.p <- c("Other"="#656565",  "Cytosolic ribosome"="#B50000", "Ethylene biosynthesis"="#EF8F6D")

tmt.BM.MP.m$value[tmt.BM.MP.m$value > 2] <- 2
tmt.BM.MP.m$value[tmt.BM.MP.m$value < -2] <- -2


ggplot(tmt.BM.MP.m, mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p, name="")+
  theme_ridges() +
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(BM/MP)")+
  xlim(-2,2)
ggsave("LeafSections_Cyto_BM_MP.jpg", width = 5, height = 6)
```

##Parent expression density curves
```{r}
tmt.B73.Mo17 <- tmt[,c("Accession","Gene","l2.B73.1.Mo17.1", "l2.B73.2.Mo17.2","l2.B73.3.Mo17.3","l2.B73.4.Mo17.4","l2.B73.5.Mo17.5")]
colnames(tmt.B73.Mo17) <- c("Accession","Gene","Section1","Section2","Section3","Section4","Section5")
tmt.B73.Mo17.m <- melt(tmt.B73.Mo17)

tmt.B73.Mo17.m$Category <- "Other"
tmt.B73.Mo17.m$Order <- 1

tmt.B73.Mo17.m$Category[tmt.B73.Mo17.m$Gene %in% GOIs$plastid] <- "Plastid-localized"
tmt.B73.Mo17.m$Order[tmt.B73.Mo17.m$Gene %in% GOIs$plastid] <- 2

tmt.B73.Mo17.m$Category[tmt.B73.Mo17.m$Gene %in% GOIs$PhANGs] <- "PhANGs"
tmt.B73.Mo17.m$Order[tmt.B73.Mo17.m$Gene %in% GOIs$PhANGs] <- 3

tmt.B73.Mo17.m$Category[tmt.B73.Mo17.m$Gene %in% GOIs$PhAPGs] <- "PhAPGs"
tmt.B73.Mo17.m$Order[tmt.B73.Mo17.m$Gene %in% GOIs$PhAPGs] <- 4

tmt.B73.Mo17.m$Category[tmt.B73.Mo17.m$Gene %in% GOIs$NE.PtRibo] <- "NE plastid ribosome"
tmt.B73.Mo17.m$Order[tmt.B73.Mo17.m$Gene %in% GOIs$NE.PtRibo] <- 5

tmt.B73.Mo17.m$Category[tmt.B73.Mo17.m$Gene %in% GOIs$PE.ribo] <- "PE plastid ribosome"
tmt.B73.Mo17.m$Order[tmt.B73.Mo17.m$Gene %in% GOIs$PE.ribo] <- 6

tmt.B73.Mo17.m$Category[tmt.B73.Mo17.m$Gene %in% GOIs$EtBio] <- "Ethylene biosynthesis"
tmt.B73.Mo17.m$Order[tmt.B73.Mo17.m$Gene %in% GOIs$EtBio] <-7

tmt.B73.Mo17.m$Category[tmt.B73.Mo17.m$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"
tmt.B73.Mo17.m$Order[tmt.B73.Mo17.m$Gene %in% GOIs$CytoRibo] <-8

colors.p <- c("Other"="#656565", "Plastid-localized"="black", "Cytosolic ribosome"="#B50000", "PE plastid ribosome"="#6A00EA", "NE plastid ribosome"="#E5A3FF", "PhAPGs"="#008275", "PhANGs"="#81E401", "Ethylene biosynthesis"="#EF8F6D")

tmt.B73.Mo17.m$value[tmt.B73.Mo17.m$value > 2] <- 2
tmt.B73.Mo17.m$value[tmt.B73.Mo17.m$value < -2] <- -2

ggplot(tmt.B73.Mo17.m[tmt.B73.Mo17.m$Category %in% c("Plastid-localized","PhANGs","PhAPGs"),], mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p[c("Plastid-localized","PhANGs","PhAPGs")], name="")+
  theme_ridges() +
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(B73/Mo17)")+
  xlim(-2,2)
ggsave("LeafSections_PhAGs_B73_Mo17.jpg", width = 5, height = 6)

ggplot(tmt.B73.Mo17.m[tmt.B73.Mo17.m$Category %in% c("Plastid-localized","NE plastid ribosome","PE plastid ribosome"),], mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p[c("Plastid-localized","NE plastid ribosome","PE plastid ribosome")], name="")+
  theme_ridges() +
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(B73/Mo17)")+
  xlim(-2,2)
ggsave("LeafSections_pRibo_B73_Mo17.jpg", width = 5, height = 6)

ggplot(tmt.B73.Mo17.m[tmt.B73.Mo17.m$Category %in% c("Other","Cytosolic ribosome","Ethylene biosynthesis"),], mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p[c("Other","Cytosolic ribosome","Ethylene biosynthesis")], name="")+
  theme_ridges() +
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(B73/Mo17)")+
  xlim(-2,2)
ggsave("LeafSections_Cyto_B73_Mo17.jpg", width = 5, height = 6)
```

##only additive proteins

```{r}
tmt.additive.up.1 <- tmt[tmt$BM.1.MP.1 > 1.1,]
tmt.additive.up.2 <- tmt[tmt$BM.2.MP.2 > 1.1,]
tmt.additive.up.3 <- tmt[tmt$BM.3.MP.3 > 1.1,]
tmt.additive.up.4 <- tmt[tmt$BM.4.MP.4 > 1.1,]
tmt.additive.up.5 <- tmt[tmt$BM.5.MP.5 > 1.1,]

tmt.additive.down.1 <- tmt[tmt$BM.1.MP.1 < 0.9,]
tmt.additive.down.2 <- tmt[tmt$BM.2.MP.2 < 0.9,]
tmt.additive.down.3 <- tmt[tmt$BM.3.MP.3 < 0.9,]
tmt.additive.down.4 <- tmt[tmt$BM.4.MP.4 < 0.9,]
tmt.additive.down.5 <- tmt[tmt$BM.5.MP.5 < 0.9,]
```
```{r}
tmt.additive.up.intersect <- tmt[tmt$Accession %in% Reduce(intersect, list(tmt.additive.up.1$Accession, tmt.additive.up.2$Accession, tmt.additive.up.3$Accession, tmt.additive.up.4$Accession, tmt.additive.up.5$Accession)),]
write.csv(tmt.additive.up.intersect[,c("Accession","Entrez","Protein")], "tmt.additive.up.intersect.csv")

tmt.additive.up.intersect.pca <- t(tmt.additive.up.intersect[,2:16])
colnames(tmt.additive.up.intersect.pca) <- tmt.additive.up.intersect$Accession
tmt.additive.up.intersect.pca <- scale(tmt.additive.up.intersect.pca, scale = T, center = F)
tmt.additive.up.intersect.pca <- prcomp(tmt.additive.up.intersect.pca)

ggbiplot(tmt.additive.up.intersect.pca, var.axes = F, labels = colnames(tmt.additive.up.intersect[,2:16]), groups = tmt.gt)+
  theme_bw()+
  theme(legend.position = "none")
ggsave("LeafSections_additive.up.intersect_PCA.png", width = 4, height = 4)
```


```{r}
atlas.cpm <- read.csv("aag1125_supportingfile_table_s1.csv", stringsAsFactors = F)
```

```{r}
NATpairs <- read.csv("NATpairs.csv", stringsAsFactors = F)
```
```{r}
colnames(NATpairs)
```

```{r}
colnames(NATpairs) <- c("sense","antisense")
```

```{r}
NATpairs.grmzm <- NATpairs[grep("GRMZM", NATpairs$sense),]
NATpairs.grmzm <- NATpairs.grmzm[grep("GRMZM", NATpairs.grmzm$antisense),]
```
```{r}
nrow(NATpairs.grmzm)
```
```{r}
atlas.dnsaf <- read.csv("aag1125_supportingfile_table_s2.csv", stringsAsFactors = F)
```


```{r}
NATpairs.grmzm.cpm <- merge.data.frame(NATpairs.grmzm, atlas.cpm[,c("tracking_id","Mature.Leaf.8")], by.x = "sense", by.y = "tracking_id", all.x = F, all.y = F)
NATpairs.grmzm.cpm <- merge.data.frame(NATpairs.grmzm.cpm, atlas.cpm[,c("tracking_id","Mature.Leaf.8")], by.x = "antisense", by.y = "tracking_id", all.x = F, all.y = F)
colnames(NATpairs.grmzm.cpm) <- c("antisense", "sense", "MLsense.cpm","MLantisense.cpm")

NATpairs.grmzm.cpm.dnsaf <- merge.data.frame(NATpairs.grmzm.cpm, atlas.dnsaf[,c("Gene.ID..Group.Leader.","Mature.Leaf.8")], by.x = "sense", by.y = "Gene.ID..Group.Leader.", all.x = T, all.y = F)
colnames(NATpairs.grmzm.cpm.dnsaf)[5] <- "MLsense.dnsaf"

NATpairs.grmzm.cpm.dnsaf <- NATpairs.grmzm.cpm.dnsaf[,c("sense","antisense","MLsense.cpm", "MLantisense.cpm", "MLsense.dnsaf")]

NATpairs.grmzm.cpm.dnsaf <- NATpairs.grmzm.cpm.dnsaf[!NATpairs.grmzm.cpm.dnsaf$MLsense.cpm==0,]
NATpairs.grmzm.cpm.dnsaf <- NATpairs.grmzm.cpm.dnsaf[!NATpairs.grmzm.cpm.dnsaf$MLantisense.cpm==0,]

NATpairs.grmzm.cpm.dnsaf$MLsense.antisense.cpm <- NATpairs.grmzm.cpm.dnsaf$MLsense.cpm/NATpairs.grmzm.cpm.dnsaf$MLantisense.cpm

NATpairs.grmzm.cpm.dnsaf.atlas <- (merge.data.frame(NATpairs.grmzm.cpm.dnsaf, atlas.cpm[,c("tracking_id", "Gene_Set", "gene_biotype", "Syntany.")], by.x = "sense", by.y = "tracking_id", all=F))

NATpairs.grmzm.cpm.dnsaf.atlas <- NATpairs.grmzm.cpm.dnsaf.atlas[order(NATpairs.grmzm.cpm.dnsaf.atlas$MLsense.antisense.cpm),]

write.csv(NATpairs.grmzm.cpm.dnsaf.atlas, "NATpairs.grmzm.MatureLeaf.cpm.dnsaf.atlas.csv", row.names = F)
```

```{r}
NATpairs.grmzm.cpm <- merge.data.frame(NATpairs.grmzm, atlas.cpm[,c("tracking_id","Primary.Root.5.Days")], by.x = "sense", by.y = "tracking_id", all.x = F, all.y = F)
NATpairs.grmzm.cpm <- merge.data.frame(NATpairs.grmzm.cpm, atlas.cpm[,c("tracking_id","Primary.Root.5.Days")], by.x = "antisense", by.y = "tracking_id", all.x = F, all.y = F)
colnames(NATpairs.grmzm.cpm) <- c("antisense", "sense", "PRsense.cpm","PRantisense.cpm")

NATpairs.grmzm.cpm.dnsaf <- merge.data.frame(NATpairs.grmzm.cpm, atlas.dnsaf[,c("Gene.ID..Group.Leader.","Primary.Root.5.Days")], by.x = "sense", by.y = "Gene.ID..Group.Leader.", all.x = T, all.y = F)
colnames(NATpairs.grmzm.cpm.dnsaf)[5] <- "PRsense.dnsaf"

NATpairs.grmzm.cpm.dnsaf <- NATpairs.grmzm.cpm.dnsaf[,c("sense","antisense","PRsense.cpm", "PRantisense.cpm", "PRsense.dnsaf")]

NATpairs.grmzm.cpm.dnsaf <- NATpairs.grmzm.cpm.dnsaf[!NATpairs.grmzm.cpm.dnsaf$PRsense.cpm==0,]
NATpairs.grmzm.cpm.dnsaf <- NATpairs.grmzm.cpm.dnsaf[!NATpairs.grmzm.cpm.dnsaf$PRantisense.cpm==0,]

NATpairs.grmzm.cpm.dnsaf$PRsense.antisense.cpm <- NATpairs.grmzm.cpm.dnsaf$PRsense.cpm/NATpairs.grmzm.cpm.dnsaf$PRantisense.cpm

NATpairs.grmzm.cpm.dnsaf.atlas <- (merge.data.frame(NATpairs.grmzm.cpm.dnsaf, atlas.cpm[,c("tracking_id", "Gene_Set", "gene_biotype", "Syntany.")], by.x = "sense", by.y = "tracking_id", all=F))

NATpairs.grmzm.cpm.dnsaf.atlas <- NATpairs.grmzm.cpm.dnsaf.atlas[order(NATpairs.grmzm.cpm.dnsaf.atlas$PRsense.antisense.cpm),]

write.csv(NATpairs.grmzm.cpm.dnsaf.atlas, "NATpairs.grmzm.PrimaryRoot.cpm.dnsaf.atlas.csv", row.names = F)
```

```{r}
NATpairs.grmzm.cpm <- merge.data.frame(NATpairs.grmzm, atlas.cpm[,c("tracking_id","Silk")], by.x = "sense", by.y = "tracking_id", all.x = F, all.y = F)
NATpairs.grmzm.cpm <- merge.data.frame(NATpairs.grmzm.cpm, atlas.cpm[,c("tracking_id","Silk")], by.x = "antisense", by.y = "tracking_id", all.x = F, all.y = F)
colnames(NATpairs.grmzm.cpm) <- c("antisense", "sense", "Silksense.cpm","Silkantisense.cpm")

NATpairs.grmzm.cpm.dnsaf <- merge.data.frame(NATpairs.grmzm.cpm, atlas.dnsaf[,c("Gene.ID..Group.Leader.","Silk..unpollinated.")], by.x = "sense", by.y = "Gene.ID..Group.Leader.", all.x = T, all.y = F)
colnames(NATpairs.grmzm.cpm.dnsaf)[5] <- "Silksense.dnsaf"

NATpairs.grmzm.cpm.dnsaf <- NATpairs.grmzm.cpm.dnsaf[,c("sense","antisense","Silksense.cpm", "Silkantisense.cpm", "Silksense.dnsaf")]

NATpairs.grmzm.cpm.dnsaf <- NATpairs.grmzm.cpm.dnsaf[!NATpairs.grmzm.cpm.dnsaf$Silksense.cpm==0,]
NATpairs.grmzm.cpm.dnsaf <- NATpairs.grmzm.cpm.dnsaf[!NATpairs.grmzm.cpm.dnsaf$Silkantisense.cpm==0,]

NATpairs.grmzm.cpm.dnsaf$Silksense.antisense.cpm <- NATpairs.grmzm.cpm.dnsaf$Silksense.cpm/NATpairs.grmzm.cpm.dnsaf$Silkantisense.cpm

NATpairs.grmzm.cpm.dnsaf.atlas <- (merge.data.frame(NATpairs.grmzm.cpm.dnsaf, atlas.cpm[,c("tracking_id", "Gene_Set", "gene_biotype", "Syntany.")], by.x = "sense", by.y = "tracking_id", all=F))

NATpairs.grmzm.cpm.dnsaf.atlas <- NATpairs.grmzm.cpm.dnsaf.atlas[order(NATpairs.grmzm.cpm.dnsaf.atlas$Silksense.antisense.cpm),]

write.csv(NATpairs.grmzm.cpm.dnsaf.atlas, "NATpairs.grmzm.Silk.cpm.dnsaf.atlas.csv", row.names = F)
```

```{r}
NATpairs.grmzm.cpm <- merge.data.frame(NATpairs.grmzm, atlas.cpm$[,c("tracking_id","Mature.Leaf.8")], by.x = "sense", by.y = "tracking_id", all.x = F, all.y = F)
NATpairs.grmzm.cpm <- merge.data.frame(NATpairs.grmzm.cpm, atlas.cpm[,c("tracking_id","Mature.Leaf.8")], by.x = "antisense", by.y = "tracking_id", all.x = F, all.y = F)
colnames(NATpairs.grmzm.cpm) <- c("antisense", "sense", "MLsense.cpm","MLantisense.cpm")

NATpairs.grmzm.cpm.dnsaf <- merge.data.frame(NATpairs.grmzm.cpm, atlas.dnsaf[,c("Gene.ID..Group.Leader.","Mature.Leaf.8")], by.x = "sense", by.y = "Gene.ID..Group.Leader.", all.x = T, all.y = F)
colnames(NATpairs.grmzm.cpm.dnsaf)[5] <- "MLsense.dnsaf"

NATpairs.grmzm.cpm.dnsaf <- NATpairs.grmzm.cpm.dnsaf[,c("sense","antisense","MLsense.cpm", "MLantisense.cpm", "MLsense.dnsaf")]

NATpairs.grmzm.cpm.dnsaf <- NATpairs.grmzm.cpm.dnsaf[!NATpairs.grmzm.cpm.dnsaf$MLsense.cpm==0,]
NATpairs.grmzm.cpm.dnsaf <- NATpairs.grmzm.cpm.dnsaf[!NATpairs.grmzm.cpm.dnsaf$MLantisense.cpm==0,]

NATpairs.grmzm.cpm.dnsaf$MLsense.antisense.cpm <- NATpairs.grmzm.cpm.dnsaf$MLsense.cpm/NATpairs.grmzm.cpm.dnsaf$MLantisense.cpm

NATpairs.grmzm.cpm.dnsaf.atlas <- (merge.data.frame(NATpairs.grmzm.cpm.dnsaf, atlas.cpm[,c("tracking_id", "Gene_Set", "gene_biotype", "Syntany.")], by.x = "sense", by.y = "tracking_id", all=F))

NATpairs.grmzm.cpm.dnsaf.atlas <- NATpairs.grmzm.cpm.dnsaf.atlas[order(NATpairs.grmzm.cpm.dnsaf.atlas$MLsense.antisense.cpm),]

write.csv(NATpairs.grmzm.cpm.dnsaf.atlas, "NATpairs.grmzm.MatureLeaf.cpm.dnsaf.atlas.csv", row.names = F)
```

```{r}
median(NATpairs.grmzm.cpm.dnsaf.atlas$MLsense.antisense.cpm[NATpairs.grmzm.cpm.dnsaf.atlas$Syntany.=="Non-Syntenic"], na.rm = T)
```

```{r}
median(NATpairs.grmzm.cpm.dnsaf.atlas$MLsense.antisense.cpm[NATpairs.grmzm.cpm.dnsaf.atlas$Syntany.=="Syntenic"], na.rm = T)
```
```{r}
NATpairs.grmzm.cpm.dnsaf.atlas[!is.na(NATpairs.grmzm.cpm.dnsaf.atlas$Syntany.),]
```

```{r}
ggplot(NATpairs.grmzm.cpm.dnsaf.atlas[!is.na(NATpairs.grmzm.cpm.dnsaf.atlas$Syntany.),], mapping = aes(x=Syntany., y=MLsense.antisense.cpm))+
  geom_violin()+
  xlab("Syntany")+
  ylab("sense/antisense cpm")
ggsave("MatureLeaf_Sense2Antisense_Syntany_Violin.png", width = 5, height = 8)
```


```{r}
NATpairs.grmzm.cpm.dnsaf.atlas[NATpairs.grmzm.cpm.dnsaf.atlas$Syntany.=="Non-Syntenic",]
```
```{r}
NATpairs.grmzm.cpm.dnsaf.atlas.cor <- NATpairs.grmzm.cpm.dnsaf.atlas
```
```{r}
NATpairs.grmzm.cpm.dnsaf.atlas.cor$MLsense.dnsaf[is.na(NATpairs.grmzm.cpm.dnsaf.atlas.cor$MLsense.dnsaf)] <- 0
```
```{r}
cor(NATpairs.grmzm.cpm.dnsaf.atlas.cor$MLsense.antisense.cpm, NATpairs.grmzm.cpm.dnsaf.atlas.cor$MLsense.dnsaf)
```


```{r}
NATpairs.grmzm.cpm.dnsaf.atlas.cor
```



```{r}
NATpairs.grmzm.cpm.dnsaf$MLsense.antisense.cpm <- NATpairs.grmzm.cpm.dnsaf$MLsense.cpm/NATpairs.grmzm.cpm.dnsaf$MLantisense.cpm
```
```{r}
cor(x=NATpairs.grmzm.cpm.dnsaf$MLsense.antisense.cpm, y = NATpairs.grmzm.cpm.dnsaf$MLsense.dnsaf)
```

```{r}
ggplot(data = NATpairs.grmzm.cpm.dnsaf, mapping = aes(x=MLsense.antisense.cpm, y=MLsense.dnsaf))+
  geom_point()
```


```{r}
tmt.additive.down.intersect <- tmt[tmt$Accession %in% Reduce(intersect, list(tmt.additive.down.1$Accession, tmt.additive.down.2$Accession, tmt.additive.down.3$Accession, tmt.additive.down.4$Accession, tmt.additive.down.5$Accession)),]
write.csv(tmt.additive.down.intersect[,c("Accession","Entrez","Protein")], "tmt.additive.down.intersect.csv")

tmt.additive.down.intersect.pca <- t(tmt.additive.down.intersect[,2:16])
colnames(tmt.additive.down.intersect.pca) <- tmt.additive.down.intersect$Accession
tmt.additive.down.intersect.pca <- scale(tmt.additive.down.intersect.pca, scale = T, center = F)
tmt.additive.down.intersect.pca <- prcomp(tmt.additive.down.intersect.pca)

ggbiplot(tmt.additive.down.intersect.pca, var.axes = F, labels = colnames(tmt.additive.down.intersect[,2:16]), groups = tmt.gt)+
  theme_bw()+
  theme(legend.position = "none")
ggsave("LeafSections_additive.down.intersect_PCA.png", width = 4, height = 4)
```

```{r}
tmt.additive.up.union <- tmt[tmt$Accession %in% unique(c(tmt.additive.up.1$Accession, tmt.additive.up.2$Accession, tmt.additive.up.3$Accession, tmt.additive.up.4$Accession, tmt.additive.up.5$Accession)),]
write.csv(tmt.additive.up.union[,c("Accession","Entrez","Protein")], "tmt.additive.up.union.csv")

tmt.additive.up.union.pca <- t(tmt.additive.up.union[,2:16])
colnames(tmt.additive.up.union.pca) <- tmt.additive.up.union$Accession
tmt.additive.up.union.pca <- scale(tmt.additive.up.union.pca, scale = T, center = F)
tmt.additive.up.union.pca <- prcomp(tmt.additive.up.union.pca)

ggbiplot(tmt.additive.up.union.pca, var.axes = F, labels = colnames(tmt.additive.up.union[,2:16]), groups = tmt.gt)+
  theme_bw()+
  theme(legend.position = "none")
ggsave("LeafSections_additive.up.union_PCA.png", width = 4, height = 4)
```

```{r}
tmt.additive.down.union <- tmt[tmt$Accession %in% unique(c(tmt.additive.down.1$Accession, tmt.additive.down.2$Accession, tmt.additive.down.3$Accession, tmt.additive.down.4$Accession, tmt.additive.down.5$Accession)),]
write.csv(tmt.additive.down.union[,c("Accession","Entrez","Protein")], "tmt.additive.down.union.csv")

tmt.additive.down.union.pca <- t(tmt.additive.down.union[,2:16])
colnames(tmt.additive.down.union.pca) <- tmt.additive.down.union$Accession
tmt.additive.down.union.pca <- scale(tmt.additive.down.union.pca, scale = T, center = F)
tmt.additive.down.union.pca <- prcomp(tmt.additive.down.union.pca)

ggbiplot(tmt.additive.down.union.pca, var.axes = F, labels = colnames(tmt.additive.down.union[,2:16]), groups = tmt.gt)+
  theme_bw()+
  theme(legend.position = "none")
ggsave("LeafSections_additive.down.union_PCA.png", width = 4, height = 4)
```






```{r}
length(tmt$Accession)
```
```{r}
length(unique(c(tmt.additive.up.1$Accession, tmt.additive.up.2$Accession, tmt.additive.up.3$Accession, tmt.additive.up.4$Accession, tmt.additive.up.5$Accession)))
```
```{r}
length(Reduce(intersect, list(tmt.additive.up.1$Accession, tmt.additive.up.2$Accession, tmt.additive.up.3$Accession, tmt.additive.up.4$Accession, tmt.additive.up.5$Accession)))
```
```{r}
length(unique(c(tmt.additive.down.1$Accession, tmt.additive.down.2$Accession, tmt.additive.down.3$Accession, tmt.additive.down.4$Accession, tmt.additive.down.5$Accession)))
```
```{r}
length(Reduce(intersect, list(tmt.additive.down.1$Accession, tmt.additive.down.2$Accession, tmt.additive.down.3$Accession, tmt.additive.down.4$Accession, tmt.additive.down.5$Accession)))
```

```{r}
additive.up <- data.frame("section"= c("S1","S2","S3","S4","S5"), "up"=c(nrow(tmt.additive.up.1), nrow(tmt.additive.up.2), nrow(tmt.additive.up.3), nrow(tmt.additive.up.4), nrow(tmt.additive.up.5)), "down"=c(nrow(tmt.additive.down.1), nrow(tmt.additive.down.2), nrow(tmt.additive.down.3), nrow(tmt.additive.down.4), nrow(tmt.additive.down.5)))
additive.up.m <- melt(additive.up)
```
```{r}
ggplot(additive.up.m, mapping = aes(x=section, y=value, fill=variable))+
  geom_col(position = "dodge")+
  theme_bw()+
  ylab("number of additive proteins")
ggsave("NumberOfAdditiveProteinsAcrossSections.png", width = 4, height = 3)
```

##only protein categories
```{r}
tmt.pca <- t(tmt[tmt$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs),2:16])
colnames(tmt.pca) <- tmt$Accession[tmt$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs)]
tmt.pca <- scale(tmt.pca, scale = T, center = F)
tmt.pca <- prcomp(tmt.pca)

ggbiplot(tmt.pca, var.axes = F, labels = colnames(tmt[,2:16]), groups = tmt.gt)+
  theme_bw()+
  theme(legend.position = "none")+
  xlim(-1.75,1.25)
ggsave("LeafSections_PhANGsPhAPGs_PCA.png", width = 4, height = 4)
```
```{r}
tmt.pca <- t(tmt[tmt$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo),2:16])
colnames(tmt.pca) <- tmt$Accession[tmt$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo)]
tmt.pca <- scale(tmt.pca, scale = T, center = F)
tmt.pca <- prcomp(tmt.pca)

ggbiplot(tmt.pca, var.axes = F, labels = colnames(tmt[,2:16]), groups = tmt.gt)+
  theme_bw()+
  theme(legend.position = "none")+
  xlim(-2,1.25)
ggsave("LeafSections_pRibo_PCA.png", width = 4, height = 4)
```
```{r}
tmt.pca <- t(tmt[tmt$Gene %in% GOIs$EtBio,2:16])
colnames(tmt.pca) <- tmt$Accession[tmt$Gene %in% GOIs$EtBio]
tmt.pca <- scale(tmt.pca, scale = T, center = F)
tmt.pca <- prcomp(tmt.pca)

ggbiplot(tmt.pca, var.axes = F, labels = colnames(tmt[,2:16]), groups = tmt.gt)+
  theme_bw()+
  theme(legend.position = "none")+
  xlim(-2,1.25)
ggsave("LeafSections_EtBio_PCA.png", width = 4, height = 4)
```








#Experiment 2
##import data


```{r import protein data}
tmt2 <- read.csv("acs-B-M-H leaf section 091322 protein groups-raw.csv", stringsAsFactors = F)
n <- nrow(tmt2)
print(c("total proteins =", n), quote = F)
#Make gene column
tmt2$Gene <- substr(tmt2$Accession, start = 1, stop = 14)
#Assign Entrez numbers
##manually assign organelle-encoded accessions
organelleEntrez <- tmt2[grep("Zema|Zeam", tmt2$Gene),]
organelleEntrez <- data.frame("Gene"=organelleEntrez$Gene, "Entrez"=NA)
##assign chloroplast accessions
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp003.1"] <- "845178"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp004.1"] <- "845232"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp008.1"] <- "845201"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp013.1"] <- "845225"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp014.1"] <- "845226"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp016.1"] <- "845236"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp017.1"] <- "845174"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp018.1"] <- "845173"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp019.1"] <- "845172"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp019.1"] <- "845172"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp021.1"] <- "845229"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp022.1"] <- "845196"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp023.1"] <- "845195"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp024.1"] <- "1466361"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp026.1"] <- "845238"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp027.1"] <- "845189"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp028.1"] <- "845190"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp029.1"] <- "845182"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp030.1"] <- "845171"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp032.1"] <- "845212"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp035.1"] <- "845175"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp036.1"] <- "845191"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp039.1"] <- "845204"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp040.1"] <- "845203"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp043.1"] <- "845198"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp044.1"] <- "845222"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp045.1"] <- "845233"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp046.1"] <- "845217"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp047.1"] <- "845176"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp048.1"] <- "845200"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp050.1"] <- "845211"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp051.1"] <- "845205"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp052.1"] <- "845192"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp053.1"] <- "845193"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp054.1"] <- "845224"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp055.1"] <- "845228"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp056.1"] <- "845223"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp057.1"] <- "845177"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp058.1"] <- "845241"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp059.1"] <- "845213"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp060.1"] <- "845214"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp061.1"] <- "845237"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp062.1"] <- "845218"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp063.1"] <- "845235"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp065.1"] <- "845216"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp067.1"] <- "845220"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp075.1"] <- "845181"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp082.1"] <- "845230"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp083.1"] <- "845185"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp084.1"] <- "845221"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp085.1"] <- "1466378"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp086.1"] <- "845183"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp087.1"] <- "1466379"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp088.1"] <- "845184"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp090.1"] <- "845188"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp091.1"] <- "845179"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp092.1"] <- "845187"
##assign mitochondrial accessions
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp008.1"] <- "4055864"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp024.1"] <- "4055860"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp032.1"] <- "4055888"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp033.1"] <- "4055938"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp036.1"] <- "4055894"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp050.1"] <- "4055956"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp051.1"] <- "4055881"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp057.1"] <- "4055973"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp081.1"] <- "4055890"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp092.1"] <- "4055837"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp097.1"] <- "4055857"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp098.1"] <- "4055896"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp113.1"] <- "4055802"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp119.1"] <- "4055928"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp129.1"] <- "4055878"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp136.1"] <- "4056004"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp159.1"] <- "4055794"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp167.1"] <- "4056007"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp179.1"] <- "4056006"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp185.1"] <- "4055892"
##add organelle entrez to mgdb-assigned Entrez numbers 
Entrez <- mgdb[,c(1,12)]
colnames(Entrez) <- c("Gene", "Entrez")
Entrez <- Entrez[!Entrez$Entrez=="",]
Entrez <- rbind(Entrez, organelleEntrez)
Entrez <- unique(Entrez)
saveRDS(Entrez, "Entrez")
#add Entrez to data
tmt2 <- join(x = tmt2, y = Entrez, by = "Gene", type="left", match="first")
#reorder columns
tmt2 <- tmt2[,c(1,23,26,24,18:22,2:13,15:17)]
#remove non-maize proteins
tmt2 <- tmt2[(grep("Zm|Zema|Zeam", tmt2$Accession)),]
n2 <- nrow(tmt2)
print(c("non-maize proteins removed =", (n-n2)), quote = F)
#remove polymorphic proteins
tmt2.pm <- tmt2[grep("only", tmt2$Accession),]
tmt2 <- tmt2[-grep("only", tmt2$Accession),]
n3 <- nrow(tmt2)
print(c("polymorphic proteins removed =", (n2-n3)), quote = F)
#remove rows with missing tmt data
tmt2 <- tmt2[!tmt2[,10]==0,]
tmt2 <- tmt2[!tmt2[,11]==0,]
tmt2 <- tmt2[!tmt2[,12]==0,]
tmt2 <- tmt2[!tmt2[,13]==0,]
tmt2 <- tmt2[!tmt2[,14]==0,]
tmt2 <- tmt2[!tmt2[,15]==0,]
tmt2 <- tmt2[!tmt2[,16]==0,]
tmt2 <- tmt2[!tmt2[,17]==0,]
tmt2 <- tmt2[!tmt2[,18]==0,]
tmt2 <- tmt2[!tmt2[,19]==0,]
tmt2 <- tmt2[!tmt2[,20]==0,]
tmt2 <- tmt2[!tmt2[,21]==0,]
tmt2 <- tmt2[!tmt2[,22]==0,]
tmt2 <- tmt2[!tmt2[,23]==0,]
tmt2 <- tmt2[!tmt2[,24]==0,]
n4 <- nrow(tmt2)
print(c("proteins with missing data removed =", (n3-n4)), quote = F)
print(c("remaining proteins =", n4), quote = F)
```


##normalization
```{r}
tmt2$acs.tot4 <- rowSums(tmt2[,c("acs.s1","acs.s2","acs.s3","acs.s4")])
tmt2$acs.tot3 <- rowSums(tmt2[,c("acs.s2","acs.s3","acs.s4")])

tmt2$B73.tot4 <- rowSums(tmt2[,c("B73.s1","B73.s2","B73.s3","B73.s4")])
tmt2$B73.tot3 <- rowSums(tmt2[,c("B73.s2","B73.s3","B73.s4")])

tmt2$Mo17.tot4 <- rowSums(tmt2[,c("Mo17.s1","Mo17.s2","Mo17.s3","Mo17.s4")])
tmt2$Mo17.tot3 <- rowSums(tmt2[,c("Mo17.s2","Mo17.s3","Mo17.s4")])

tmt2$BM.tot3 <- rowSums(tmt2[,c("BM.s2","BM.s3","BM.s4")])
```
```{r}
tmt2$acs.s1.tot4 <- tmt2$acs.s1/tmt2$acs.tot4
tmt2$acs.s2.tot4 <- tmt2$acs.s2/tmt2$acs.tot4
tmt2$acs.s3.tot4 <- tmt2$acs.s3/tmt2$acs.tot4
tmt2$acs.s4.tot4 <- tmt2$acs.s4/tmt2$acs.tot4

tmt2$acs.s2.tot3 <- tmt2$acs.s2/tmt2$acs.tot3
tmt2$acs.s3.tot3 <- tmt2$acs.s3/tmt2$acs.tot3
tmt2$acs.s4.tot3 <- tmt2$acs.s4/tmt2$acs.tot3

tmt2$B73.s1.tot4 <- tmt2$B73.s1/tmt2$B73.tot4
tmt2$B73.s2.tot4 <- tmt2$B73.s2/tmt2$B73.tot4
tmt2$B73.s3.tot4 <- tmt2$B73.s3/tmt2$B73.tot4
tmt2$B73.s4.tot4 <- tmt2$B73.s4/tmt2$B73.tot4

tmt2$B73.s2.tot3 <- tmt2$B73.s2/tmt2$B73.tot3
tmt2$B73.s3.tot3 <- tmt2$B73.s3/tmt2$B73.tot3
tmt2$B73.s4.tot3 <- tmt2$B73.s4/tmt2$B73.tot3

tmt2$Mo17.s1.tot4 <- tmt2$Mo17.s1/tmt2$Mo17.tot4
tmt2$Mo17.s2.tot4 <- tmt2$Mo17.s2/tmt2$Mo17.tot4
tmt2$Mo17.s3.tot4 <- tmt2$Mo17.s3/tmt2$Mo17.tot4
tmt2$Mo17.s4.tot4 <- tmt2$Mo17.s4/tmt2$Mo17.tot4

tmt2$Mo17.s2.tot3 <- tmt2$Mo17.s2/tmt2$Mo17.tot3
tmt2$Mo17.s3.tot3 <- tmt2$Mo17.s3/tmt2$Mo17.tot3
tmt2$Mo17.s4.tot3 <- tmt2$Mo17.s4/tmt2$Mo17.tot3

tmt2$BM.s2.tot3 <- tmt2$BM.s2/tmt2$BM.tot3
tmt2$BM.s3.tot3 <- tmt2$BM.s3/tmt2$BM.tot3
tmt2$BM.s4.tot3 <- tmt2$BM.s4/tmt2$BM.tot3
```
##heatmap
```{r}
library(RColorBrewer)
```
```{r}
tmt2.m <- as.matrix(tmt2[,c("acs.s2.tot3", "acs.s3.tot3", "acs.s4.tot3", "B73.s2.tot3", "B73.s3.tot3", "B73.s4.tot3", "Mo17.s2.tot3", "Mo17.s3.tot3", "Mo17.s4.tot3", "BM.s2.tot3", "BM.s3.tot3", "BM.s4.tot3")])

colnames(tmt2.m) <- c("acs.s2", "acs.s3", "acs.s4", "B73.s2", "B73.s3", "B73.s4", "Mo17.s2", "Mo17.s3", "Mo17.s4", "BM.s2", "BM.s3", "BM.s4")

heatmap(tmt2.m, scale = "none", labRow = "")
```
```{r}
tmt2.m <- as.matrix(tmt2[,c(59:61,65:68)])

colnames(tmt2.m) <- c("BM.MP.s2", "BM.MP.s3", "BM.MP.s4", "acs.B73.s1", "acs.B73.s2", "acs.B73.s3", "acs.B73.s4")

heatmap(tmt2.m, scale = "row", labRow = "", col=cm.colors(10), cexCol = 1)

```


```{r}
tmt2.digenomic.m <- as.matrix(tmt2[tmt2$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs, GOIs$NE.PtRibo, GOIs$PE.ribo),c("acs.s2.tot3", "acs.s3.tot3", "acs.s4.tot3", "B73.s2.tot3", "B73.s3.tot3", "B73.s4.tot3", "Mo17.s2.tot3", "Mo17.s3.tot3", "Mo17.s4.tot3", "BM.s2.tot3", "BM.s3.tot3", "BM.s4.tot3")])

colnames(tmt2.digenomic.m) <- c("acs.s2", "acs.s3", "acs.s4", "B73.s2", "B73.s3", "B73.s4", "Mo17.s2", "Mo17.s3", "Mo17.s4", "BM.s2", "BM.s3", "BM.s4")

heatmap(tmt2.digenomic.m, scale = "none", labRow = "")
```
```{r CorFun}
#using dataframe and vector as input
CorFun <- function(x, cols1, cols2){
  cv <- vector(length = nrow(x))
  for(i in 1:nrow(x)){
    v1 <- as.numeric(x[i,cols1])
    v2 <- as.numeric(x[i,cols2])
    c <- cor(v1, v2, method = "pearson")
    cv[i]<- c
  }
  cv
}
```

```{r}
tmt2$cor.acs.B73 <- CorFun(tmt2, 11:13, 15:17)
tmt2$cor.acs.Mo17 <- CorFun(tmt2, 11:13, 19:21)
tmt2$cor.acs.BM <- CorFun(tmt2, 11:13, 22:24)
tmt2$cor.B73.BM <- CorFun(tmt2,  15:17,22:24)
```
```{r}
mean(tmt2[,"cor.acs.B73"])
mean(tmt2[,"cor.acs.Mo17"])
mean(tmt2[,"cor.acs.BM"])
```
```{r}
mean(tmt2[tmt2$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs, GOIs$NE.PtRibo, GOIs$PE.ribo),"cor.acs.B73"])
mean(tmt2[tmt2$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs, GOIs$NE.PtRibo, GOIs$PE.ribo),"cor.acs.Mo17"])
mean(tmt2[tmt2$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs, GOIs$NE.PtRibo, GOIs$PE.ribo),"cor.acs.BM"])
```

```{r}
tmt2.cor.acs2BM.NOTacs2B73 <- tmt2[tmt2$cor.acs.BM > 0.5 & tmt2$cor.acs.B73 < 0.5 & tmt2$cor.B73.BM < 0.5,]
tmt2[tmt2$cor.acs.BM > 0.5 & tmt2$cor.acs.B73 < 0.5 & tmt2$cor.B73.BM < 0.5,"Protein"]
```



```{r}
write.csv(na.omit(tmt2[,c(3,1,2,4:9,70:73,10:24)]), "tmt2.cor.all.csv", row.names = F)

write.csv(na.omit(tmt2.cor.acs2BM.NOTacs2B73[,c(3,1,2,4:9,70:73,10:24)]), "tmt2.cor.acs2BM.NOTacs2B73.csv", row.names = F)
```



##calculate ratios between sections
```{r}
#???
```
##plot ratios between sections
```{r tmt2.s2.tot.s1.tot}
tmt2.s2.tot.s1.tot <- tmt2[,c("Accession","Gene","l2.acs.s2.tot.acs.s1.tot", "l2.B73.s2.tot.B73.s1.tot","l2.Mo17.s2.tot.Mo17.s1.tot")]
colnames(tmt2.s2.tot.s1.tot) <- c("Accession","Gene","acs","B73","Mo17")
tmt2.s2.tot.s1.tot.m <- melt(tmt2.s2.tot.s1.tot)

tmt2.s2.tot.s1.tot.m$Category <- "Other"

tmt2.s2.tot.s1.tot.m$Category[tmt2.s2.tot.s1.tot.m$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs)] <- "PhANGs/PhAPGs"

tmt2.s2.tot.s1.tot.m$Category[tmt2.s2.tot.s1.tot.m$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo)] <- "Plastid ribosome"

tmt2.s2.tot.s1.tot.m$Category[tmt2.s2.tot.s1.tot.m$Gene %in% GOIs$EtBio] <- "Ethylene biosynthesis"

tmt2.s2.tot.s1.tot.m$Category[tmt2.s2.tot.s1.tot.m$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"

colors.p2 <- c("Other"="grey50",  "Plastid ribosome"="#6A00EA",  "PhANGs/PhAPGs"="#008275", "Ethylene biosynthesis"="#EF8F6D", "Cytosolic ribosome"="#B50000")

tmt2.s2.tot.s1.tot.m$value[tmt2.s2.tot.s1.tot.m$value <= -5] <- -5
tmt2.s2.tot.s1.tot.m$value[tmt2.s2.tot.s1.tot.m$value >= 5] <- 5

ggplot(tmt2.s2.tot.s1.tot.m, mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p2, name="")+
  theme_ridges() +
  xlim(-5,5)+
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(Section2/Section1)")+
  guides(col=guide_legend(nrow = 3))+
  coord_cartesian(clip = "off")
ggsave("tmt2.s2.tot.s1.tot.png", width = 5, height = 3)
```
```{r tmt2.s3.s2}
tmt2.s3.tot.s2.tot <- tmt2[,c("Accession","Gene","l2.acs.s3.tot.acs.s2.tot", "l2.B73.s3.tot.B73.s2.tot","l2.Mo17.s3.tot.Mo17.s2.tot", "l2.BM.s3.tot.BM.s2.tot")]
colnames(tmt2.s3.tot.s2.tot) <- c("Accession","Gene","acs","B73","Mo17","BM")
tmt2.s3.tot.s2.tot.m <- melt(tmt2.s3.tot.s2.tot)

tmt2.s3.tot.s2.tot.m$Category <- "Other"

tmt2.s3.tot.s2.tot.m$Category[tmt2.s3.tot.s2.tot.m$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs)] <- "PhANGs/PhAPGs"

tmt2.s3.tot.s2.tot.m$Category[tmt2.s3.tot.s2.tot.m$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo)] <- "Plastid ribosome"

tmt2.s3.tot.s2.tot.m$Category[tmt2.s3.tot.s2.tot.m$Gene %in% GOIs$EtBio] <- "Ethylene biosynthesis"

tmt2.s3.tot.s2.tot.m$Category[tmt2.s3.tot.s2.tot.m$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"

colors.p2 <- c("Other"="grey50",  "Plastid ribosome"="#6A00EA",  "PhANGs/PhAPGs"="#008275", "Ethylene biosynthesis"="#EF8F6D", "Cytosolic ribosome"="#B50000")

tmt2.s3.tot.s2.tot.m$value[tmt2.s3.tot.s2.tot.m$value <= -5] <- -5
tmt2.s3.tot.s2.tot.m$value[tmt2.s3.tot.s2.tot.m$value >= 5] <- 5

ggplot(tmt2.s3.tot.s2.tot.m, mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p2, name="")+
  theme_ridges() +
  xlim(-5,5)+
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(Section3/Section2)")+
  guides(col=guide_legend(nrow = 3))+
  coord_cartesian(clip = "off")
ggsave("tmt2.s3.tot.s2.tot.png", width = 5, height = 3)
```
```{r tmt2.s4.tot.s3.tot}
tmt2.s4.tot.s3.tot <- tmt2[,c("Accession","Gene","l2.acs.s4.tot.acs.s3.tot", "l2.B73.s4.tot.B73.s3.tot","l2.Mo17.s4.tot.Mo17.s3.tot", "l2.BM.s4.tot.BM.s3.tot")]
colnames(tmt2.s4.tot.s3.tot) <- c("Accession","Gene","acs","B73","Mo17","BM")
tmt2.s4.tot.s3.tot.m <- melt(tmt2.s4.tot.s3.tot)

tmt2.s4.tot.s3.tot.m$Category <- "Other"

tmt2.s4.tot.s3.tot.m$Category[tmt2.s4.tot.s3.tot.m$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs)] <- "PhANGs/PhAPGs"

tmt2.s4.tot.s3.tot.m$Category[tmt2.s4.tot.s3.tot.m$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo)] <- "Plastid ribosome"

tmt2.s4.tot.s3.tot.m$Category[tmt2.s4.tot.s3.tot.m$Gene %in% GOIs$EtBio] <- "Ethylene biosynthesis"

tmt2.s4.tot.s3.tot.m$Category[tmt2.s4.tot.s3.tot.m$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"

colors.p2 <- c("Other"="grey50",  "Plastid ribosome"="#6A00EA",  "PhANGs/PhAPGs"="#008275", "Ethylene biosynthesis"="#EF8F6D", "Cytosolic ribosome"="#B50000")

tmt2.s4.tot.s3.tot.m$value[tmt2.s4.tot.s3.tot.m$value <= -5] <- -5
tmt2.s4.tot.s3.tot.m$value[tmt2.s4.tot.s3.tot.m$value >= 5] <- 5

ggplot(tmt2.s4.tot.s3.tot.m, mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p2, name="")+
  theme_ridges() +
  xlim(-5,5)+
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(Section4/Section3)")+
  guides(col=guide_legend(nrow = 3))+
  coord_cartesian(clip = "off")
ggsave("tmt2.s4.tot.s3.tot.png", width = 5, height = 3)
```

##calculate ratios between genotypes
```{r}
tmt2$MP.s2.tot3 <- rowMeans(tmt2[,c("B73.s2.tot3", "Mo17.s2.tot3")])
tmt2$MP.s3.tot3 <- rowMeans(tmt2[,c("B73.s3.tot3", "Mo17.s3.tot3")])
tmt2$MP.s4.tot3 <- rowMeans(tmt2[,c("B73.s4.tot3", "Mo17.s4.tot3")])

tmt2$l2_BM.s2.tot3_MP.s2.tot3 <- log2(tmt2$BM.s2.tot3/tmt2$MP.s2.tot3)
tmt2$l2_BM.s3.tot3_MP.s3.tot3 <- log2(tmt2$BM.s3.tot3/tmt2$MP.s3.tot3)
tmt2$l2_BM.s4.tot3_MP.s4.tot3 <- log2(tmt2$BM.s4.tot3/tmt2$MP.s4.tot3)

tmt2$l2_BM.s2.tot3_B73.s2.tot3 <- log2(tmt2$BM.s2.tot3/tmt2$B73.s2.tot3)
tmt2$l2_BM.s3.tot3_B73.s3.tot3 <- log2(tmt2$BM.s3.tot3/tmt2$B73.s3.tot3)
tmt2$l2_BM.s4.tot3_B73.s4.tot3 <- log2(tmt2$BM.s4.tot3/tmt2$B73.s4.tot3)

tmt2$l2_acs.s1.tot4_B73.s1.tot4 <- log2(tmt2$acs.s1.tot4/tmt2$B73.s1.tot4)
tmt2$l2_acs.s2.tot4_B73.s2.tot4 <- log2(tmt2$acs.s2.tot4/tmt2$B73.s2.tot4)
tmt2$l2_acs.s3.tot4_B73.s3.tot4 <- log2(tmt2$acs.s3.tot4/tmt2$B73.s3.tot4)
tmt2$l2_acs.s4.tot4_B73.s4.tot4 <- log2(tmt2$acs.s4.tot4/tmt2$B73.s4.tot4)

tmt2$l2_NA <- 0
```


```{r tmt2.BM.MP}
tmt2.BM.MP <- tmt2[,c("Accession","Gene","l2_NA", "l2_BM.s2.tot3_MP.s2.tot3", "l2_BM.s3.tot3_MP.s3.tot3","l2_BM.s4.tot3_MP.s4.tot3")]
colnames(tmt2.BM.MP) <- c("Accession","Gene","Section1", "Section2","Section3","Section4")
tmt2.BM.MP.m <- melt(tmt2.BM.MP)

tmt2.BM.MP.m$Category <- "Other"

tmt2.BM.MP.m$Category[tmt2.BM.MP.m$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs)] <- "PhANGs/PhAPGs"

tmt2.BM.MP.m$Category[tmt2.BM.MP.m$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo)] <- "Plastid ribosome"

tmt2.BM.MP.m$Category[tmt2.BM.MP.m$Gene %in% GOIs$EtBio] <- "Ethylene biosynthesis"

tmt2.BM.MP.m$Category[tmt2.BM.MP.m$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"

tmt2.BM.MP.m$Category[tmt2.BM.MP.m$variable=="Section1"] <- "none"

colors.p2 <- c("Other"="grey50",  "Plastid ribosome"="#6A00EA",  "PhANGs/PhAPGs"="#008275", "Ethylene biosynthesis"="#EF8F6D", "Cytosolic ribosome"="#B50000")

tmt2.BM.MP.m$value[tmt2.BM.MP.m$value <= -1.5] <- -1.5
tmt2.BM.MP.m$value[tmt2.BM.MP.m$value >= 1.5] <- 1.5

ggplot(tmt2.BM.MP.m, mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p2, name="", na.value = "white")+
  theme_ridges() +
  xlim(-1.5,1.5)+
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(BM/MP)")+
  guides(col=guide_legend(nrow = 3))+
  coord_cartesian(clip = "off", expand = F)
ggsave("tmt2.BM.MP.png", width = 5, height = 4)
```

```{r tmt2.BM.B73}
tmt2.BM.B73 <- tmt2[,c("Accession","Gene","l2_NA", "l2_BM.s2.tot3_B73.s2.tot3", "l2_BM.s3.tot3_B73.s3.tot3","l2_BM.s4.tot3_B73.s4.tot3")]
colnames(tmt2.BM.B73) <- c("Accession","Gene","Section1", "Section2","Section3","Section4")
tmt2.BM.B73.m <- melt(tmt2.BM.B73)

tmt2.BM.B73.m$Category <- "Other"

tmt2.BM.B73.m$Category[tmt2.BM.B73.m$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs)] <- "PhANGs/PhAPGs"

tmt2.BM.B73.m$Category[tmt2.BM.B73.m$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo)] <- "Plastid ribosome"

tmt2.BM.B73.m$Category[tmt2.BM.B73.m$Gene %in% GOIs$EtBio] <- "Ethylene biosynthesis"

tmt2.BM.B73.m$Category[tmt2.BM.B73.m$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"

tmt2.BM.B73.m$Category[tmt2.BM.B73.m$variable=="Section1"] <- "none"

colors.p2 <- c("Other"="grey50",  "Plastid ribosome"="#6A00EA",  "PhANGs/PhAPGs"="#008275", "Ethylene biosynthesis"="#EF8F6D", "Cytosolic ribosome"="#B50000")

tmt2.BM.B73.m$value[tmt2.BM.B73.m$value <= -1.5] <- -1.5
tmt2.BM.B73.m$value[tmt2.BM.B73.m$value >= 1.5] <- 1.5

ggplot(tmt2.BM.B73.m, mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p2, name="", na.value = "white")+
  theme_ridges() +
  xlim(-1.5,1.5)+
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(BM/B73)")+
  guides(col=guide_legend(nrow = 3))+
  coord_cartesian(clip = "off", expand = F)
ggsave("tmt2.BM.B73.png", width = 5, height = 4)
```

```{r tmt2.acs.B73}
tmt2.acs.B73 <- tmt2[,c("Accession","Gene","l2_acs.s1.tot4_B73.s1.tot4","l2_acs.s2.tot4_B73.s2.tot4", "l2_acs.s3.tot4_B73.s3.tot4","l2_acs.s4.tot4_B73.s4.tot4")]
colnames(tmt2.acs.B73) <- c("Accession","Gene","Section1", "Section2","Section3","Section4")
tmt2.acs.B73.m <- melt(tmt2.acs.B73)

tmt2.acs.B73.m$Category <- "Other"

tmt2.acs.B73.m$Category[tmt2.acs.B73.m$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs)] <- "PhANGs/PhAPGs"

tmt2.acs.B73.m$Category[tmt2.acs.B73.m$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo)] <- "Plastid ribosome"

tmt2.acs.B73.m$Category[tmt2.acs.B73.m$Gene %in% GOIs$EtBio] <- "Ethylene biosynthesis"

tmt2.acs.B73.m$Category[tmt2.acs.B73.m$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"

colors.p2 <- c("Other"="grey50",  "Plastid ribosome"="#6A00EA",  "PhANGs/PhAPGs"="#008275", "Ethylene biosynthesis"="#EF8F6D", "Cytosolic ribosome"="#B50000")

tmt2.acs.B73.m$value[tmt2.acs.B73.m$value <= -1.5] <- -1.5
tmt2.acs.B73.m$value[tmt2.acs.B73.m$value >= 1.5] <- 1.5

ggplot(tmt2.acs.B73.m, mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p2, name="")+
  theme_ridges() +
  xlim(-1.5,1.5)+
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(acs/B73)")+
  guides(col=guide_legend(nrow = 3))+
  coord_cartesian(clip = "off", expand = F)
ggsave("tmt2.acs.B73.png", width = 5, height = 4)
```


##PCA
```{r}
colnames(tmt2)
```

```{r}
tmt2.pca <- t(tmt2[,c("acs.s2.tot3","acs.s3.tot3","acs.s4.tot3","B73.s2.tot3","B73.s3.tot3","B73.s4.tot3","Mo17.s2.tot3","Mo17.s3.tot3","Mo17.s4.tot3", "BM.s2.tot3","BM.s3.tot3","BM.s4.tot3")])
colnames(tmt2.pca) <- tmt2$Accession
tmt2.pca <- scale(tmt2.pca, scale = T, center = F)
tmt2.pca <- prcomp(tmt2.pca)

tmt2.gt <- c(rep("acs",3), rep("B73",3), rep("Mo17",3), rep("BxM",3))


ggbiplot(tmt2.pca, choices = 1:2, var.axes = F, labels = c("acs.s2","acs.s3","acs.s4","B73.s2","B73.s3","B73.s4","Mo17.s2","Mo17.s3","Mo17.s4","BM.s2","BM.s3","BM.s4"), groups = tmt2.gt)+
  theme_bw()+
  theme(legend.position = "none")
```
```{r}
tmt2.pca.x <- as.data.frame(tmt2.pca$x)
tmt2.pca.x$sample <- c("acs.s2","acs.s3","acs.s4","B73.s2","B73.s3","B73.s4","Mo17.s2","Mo17.s3","Mo17.s4","BM.s2","BM.s3","BM.s4")
tmt2.pca.x$section <- rep(c("s2","s3","s4"),4)
tmt2.pca.x$genotype <- c(rep("acs",3), rep("B73",3),rep("Mo17",3), rep("BM",3))
```
```{r}
ggplot(tmt2.pca.x, mapping = aes(x = section, y = PC1, color = genotype)) +
  scale_color_manual(values = c("acs"="#6fa8dc","B73"="#e06666","Mo17"="#ffd966","BM"="#e69138"))+
  geom_point(size=4)+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))
```

```{r}
plotlist <- list()
for (i in 1:8) {
  df <- data.frame("value"=tmt2.pca.x[,i], "sample"=c("acs.s2","acs.s3","acs.s4","B73.s2","B73.s3","B73.s4","Mo17.s2","Mo17.s3","Mo17.s4","BM.s2","BM.s3","BM.s4"), "genotype"=c(rep("acs",3), rep("B73",3),rep("Mo17",3), rep("BM",3)), "section"=rep(c("s2","s3","s4"),4))

plotlist[[i]] <- ggplot(df, mapping = aes(x = section, y = value, color = genotype)) +
  scale_color_manual(values = c("acs"="#6fa8dc","B73"="#e06666","Mo17"="#ffd966","BM"="#e69138"))+
  geom_point(size=4)+
  theme_bw()+
  geom_hline(yintercept = 0)+
  ggtitle(paste0("PC",i))
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))
}
plotlist
```


#Experiment 3
```{r}
tmt3 <- read.csv("B-M-H-A leaf section 021323-protein groups.csv", stringsAsFactors = F)
```
```{r}
tmt3 <- tmt3[,c(17:22,24,1:16)]
#tmt3 <- tmt3[,c(17:22,24,25:40)]

colnames(tmt3)[7] <- "Protein"

tmt3$Gene <- substr(tmt3$Accession, start = 1, stop = 14)

tmt3 <- tmt3[,c(24,1:23)]
```
```{r}
##manually assign organelle-encoded accessions
organelleEntrez <- tmt3[grep("Zema|Zeam", tmt3$Gene),]
organelleEntrez <- data.frame("Gene"=organelleEntrez$Gene, "Entrez"=NA)
##assign chloroplast accessions
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp003.1"] <- "845178"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp004.1"] <- "845232"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp008.1"] <- "845201"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp013.1"] <- "845225"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp014.1"] <- "845226"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp016.1"] <- "845236"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp017.1"] <- "845174"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp018.1"] <- "845173"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp019.1"] <- "845172"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp019.1"] <- "845172"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp021.1"] <- "845229"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp022.1"] <- "845196"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp023.1"] <- "845195"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp024.1"] <- "1466361"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp026.1"] <- "845238"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp027.1"] <- "845189"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp028.1"] <- "845190"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp029.1"] <- "845182"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp030.1"] <- "845171"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp032.1"] <- "845212"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp035.1"] <- "845175"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp036.1"] <- "845191"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp039.1"] <- "845204"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp040.1"] <- "845203"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp043.1"] <- "845198"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp044.1"] <- "845222"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp045.1"] <- "845233"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp046.1"] <- "845217"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp047.1"] <- "845176"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp048.1"] <- "845200"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp050.1"] <- "845211"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp051.1"] <- "845205"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp052.1"] <- "845192"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp053.1"] <- "845193"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp054.1"] <- "845224"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp055.1"] <- "845228"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp056.1"] <- "845223"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp057.1"] <- "845177"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp058.1"] <- "845241"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp059.1"] <- "845213"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp060.1"] <- "845214"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp061.1"] <- "845237"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp062.1"] <- "845218"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp063.1"] <- "845235"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp065.1"] <- "845216"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp067.1"] <- "845220"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp075.1"] <- "845181"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp082.1"] <- "845230"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp083.1"] <- "845185"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp084.1"] <- "845221"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp085.1"] <- "1466378"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp086.1"] <- "845183"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp087.1"] <- "1466379"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp088.1"] <- "845184"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp090.1"] <- "845188"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp091.1"] <- "845179"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp092.1"] <- "845187"
##assign mitochondrial accessions
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp008.1"] <- "4055864"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp024.1"] <- "4055860"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp032.1"] <- "4055888"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp033.1"] <- "4055938"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp036.1"] <- "4055894"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp050.1"] <- "4055956"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp051.1"] <- "4055881"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp057.1"] <- "4055973"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp081.1"] <- "4055890"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp092.1"] <- "4055837"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp097.1"] <- "4055857"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp098.1"] <- "4055896"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp113.1"] <- "4055802"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp119.1"] <- "4055928"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp129.1"] <- "4055878"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp136.1"] <- "4056004"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp159.1"] <- "4055794"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp167.1"] <- "4056007"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp179.1"] <- "4056006"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp185.1"] <- "4055892"
##add organelle entrez to mgdb-assigned Entrez numbers 
Entrez <- mgdb[,c(1,12)]
colnames(Entrez) <- c("Gene", "Entrez")
Entrez <- Entrez[!Entrez$Entrez=="",]
Entrez <- rbind(Entrez, organelleEntrez)
Entrez <- unique(Entrez)
saveRDS(Entrez, "Entrez")
```
```{r}
tmt3 <- join(x = tmt3, y = Entrez, by = "Gene", type="left", match="first")

tmt3 <- tmt3[,c(25,1:24)]
colnames(tmt3)
```
```{r}
n <- nrow(tmt3)
print(c("total proteins =", n), quote = F)
#remove non-maize proteins
tmt3 <- tmt3[(grep("Zm|Zema|Zeam", tmt3$Accession)),]
n2 <- nrow(tmt3)
print(c("non-maize proteins removed =", (n-n2)), quote = F)
#remove polymorphic proteins
tmt3.pm <- tmt3[grep("only", tmt3$Accession),]
tmt3 <- tmt3[-grep("only", tmt3$Accession),]
n3 <- nrow(tmt3)
print(c("polymorphic proteins removed =", (n2-n3)), quote = F)
#remove rows with missing tmt data
tmt3 <- tmt3[!tmt3[,10]==0,]
tmt3 <- tmt3[!tmt3[,11]==0,]
tmt3 <- tmt3[!tmt3[,12]==0,]
tmt3 <- tmt3[!tmt3[,13]==0,]
tmt3 <- tmt3[!tmt3[,14]==0,]
tmt3 <- tmt3[!tmt3[,15]==0,]
tmt3 <- tmt3[!tmt3[,16]==0,]
tmt3 <- tmt3[!tmt3[,17]==0,]
tmt3 <- tmt3[!tmt3[,18]==0,]
tmt3 <- tmt3[!tmt3[,19]==0,]
tmt3 <- tmt3[!tmt3[,20]==0,]
tmt3 <- tmt3[!tmt3[,21]==0,]
tmt3 <- tmt3[!tmt3[,22]==0,]
tmt3 <- tmt3[!tmt3[,23]==0,]
tmt3 <- tmt3[!tmt3[,24]==0,]
tmt3 <- tmt3[!tmt3[,25]==0,]
n4 <- nrow(tmt3)
print(c("proteins with missing data removed =", (n3-n4)), quote = F)
print(c("remaining proteins =", n4), quote = F)
```
```{r calculate ratios}
tmt3$MP.1 <- rowMeans(tmt3[,c("B.1","M.1")])
tmt3$MP.2 <- rowMeans(tmt3[,c("B.2","M.2")])
tmt3$MP.3 <- rowMeans(tmt3[,c("B.3","M.3")])
tmt3$MP.4 <- rowMeans(tmt3[,c("B.4","M.4")])

tmt3$l2.H1.MP1 <- log2(tmt3$H.1/tmt3$MP.1)
tmt3$l2.H2.MP2 <- log2(tmt3$H.2/tmt3$MP.2)
tmt3$l2.H3.MP3 <- log2(tmt3$H.3/tmt3$MP.3)
tmt3$l2.H4.MP4 <- log2(tmt3$H.4/tmt3$MP.4)

tmt3$l2.A1.B1 <- log2(tmt3$A.1/tmt3$B.1)
tmt3$l2.A2.B2 <- log2(tmt3$A.2/tmt3$B.2)
tmt3$l2.A3.B3 <- log2(tmt3$A.3/tmt3$B.3)
tmt3$l2.A4.B4 <- log2(tmt3$A.4/tmt3$B.4)
```
```{r}
tmt3.BM.MP <- tmt3[,c("Accession","Gene","l2.H1.MP1", "l2.H2.MP2", "l2.H3.MP3","l2.H4.MP4")]
colnames(tmt3.BM.MP) <- c("Accession","Gene","Section1", "Section2","Section3","Section4")
tmt3.BM.MP.m <- melt(tmt3.BM.MP)

tmt3.BM.MP.m$Category <- "Other"

tmt3.BM.MP.m$Category[tmt3.BM.MP.m$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs)] <- "PhANGs/PhAPGs"

tmt3.BM.MP.m$Category[tmt3.BM.MP.m$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo)] <- "Plastid ribosome"


tmt3.BM.MP.m$Category[tmt3.BM.MP.m$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"


colors.p2 <- c("Other"="grey50",  "Plastid ribosome"="#6A00EA",  "PhANGs/PhAPGs"="#008275", "Cytosolic ribosome"="#B50000")

tmt3.BM.MP.m$value[tmt3.BM.MP.m$value <= -3] <- -3
tmt3.BM.MP.m$value[tmt3.BM.MP.m$value >= 3] <- 3

ggplot(tmt3.BM.MP.m, mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p2, name="")+
  theme_ridges() +
  xlim(-3,3)+
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(BM/MP)")+
  guides(col=guide_legend(nrow = 3))+
  coord_cartesian(clip = "off", expand = F)
ggsave("tmt3.BM.MP.png", width = 4, height = 5)
```
```{r tmt3.acs.B73}
tmt3.acs.B73 <- tmt3[,c("Accession","Gene","l2.A1.B1","l2.A2.B2", "l2.A3.B3","l2.A4.B4")]
colnames(tmt3.acs.B73) <- c("Accession","Gene","Section1", "Section2","Section3","Section4")
tmt3.acs.B73.m <- melt(tmt3.acs.B73)

tmt3.acs.B73.m$Category <- "Other"

tmt3.acs.B73.m$Category[tmt3.acs.B73.m$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs)] <- "PhANGs/PhAPGs"

tmt3.acs.B73.m$Category[tmt3.acs.B73.m$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo)] <- "Plastid ribosome"


tmt3.acs.B73.m$Category[tmt3.acs.B73.m$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"

colors.p2 <- c("Other"="grey50",  "Plastid ribosome"="#6A00EA",  "PhANGs/PhAPGs"="#008275", "Cytosolic ribosome"="#B50000")

tmt3.acs.B73.m$value[tmt3.acs.B73.m$value <= -3] <- -3
tmt3.acs.B73.m$value[tmt3.acs.B73.m$value >= 3] <- 3

ggplot(tmt3.acs.B73.m, mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p2, name="")+
  theme_ridges() +
  xlim(-3,3)+
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(acs/B73)")+
  guides(col=guide_legend(nrow = 3))+
  coord_cartesian(clip = "off", expand = F)
ggsave("tmt3.acs.B73.png", width = 4, height = 5)
```
#Experiment 4
```{r import data}
tmt4 <- read.csv("B-M-H-A leaf section 050123.csv", stringsAsFactors = F)

tmt4 <- tmt4[,c(17:23,1:16)]

colnames(tmt4)[8:23] <- c("B.1","B.2","B.3","B.4","M.1","M.2","M.3","M.4","H.1","H.2","H.3","H.4","A.1","A.2","A.3","A.4")

tmt4$Gene <- substr(tmt4$Accession, start = 1, stop = 14)

tmt4 <- tmt4[,c(24,1:23)]
colnames(tmt4)
```
```{r entrez}
##manually assign organelle-encoded accessions
organelleEntrez <- tmt4[grep("Zema|Zeam", tmt4$Gene),]
organelleEntrez <- data.frame("Gene"=organelleEntrez$Gene, "Entrez"=NA)
##assign chloroplast accessions
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp003.1"] <- "845178"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp004.1"] <- "845232"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp008.1"] <- "845201"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp013.1"] <- "845225"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp014.1"] <- "845226"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp016.1"] <- "845236"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp017.1"] <- "845174"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp018.1"] <- "845173"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp019.1"] <- "845172"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp019.1"] <- "845172"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp021.1"] <- "845229"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp022.1"] <- "845196"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp023.1"] <- "845195"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp024.1"] <- "1466361"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp026.1"] <- "845238"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp027.1"] <- "845189"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp028.1"] <- "845190"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp029.1"] <- "845182"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp030.1"] <- "845171"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp032.1"] <- "845212"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp035.1"] <- "845175"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp036.1"] <- "845191"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp039.1"] <- "845204"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp040.1"] <- "845203"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp043.1"] <- "845198"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp044.1"] <- "845222"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp045.1"] <- "845233"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp046.1"] <- "845217"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp047.1"] <- "845176"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp048.1"] <- "845200"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp050.1"] <- "845211"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp051.1"] <- "845205"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp052.1"] <- "845192"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp053.1"] <- "845193"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp054.1"] <- "845224"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp055.1"] <- "845228"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp056.1"] <- "845223"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp057.1"] <- "845177"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp058.1"] <- "845241"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp059.1"] <- "845213"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp060.1"] <- "845214"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp061.1"] <- "845237"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp062.1"] <- "845218"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp063.1"] <- "845235"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp065.1"] <- "845216"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp067.1"] <- "845220"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp075.1"] <- "845181"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp082.1"] <- "845230"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp083.1"] <- "845185"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp084.1"] <- "845221"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp085.1"] <- "1466378"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp086.1"] <- "845183"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp087.1"] <- "1466379"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp088.1"] <- "845184"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp090.1"] <- "845188"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp091.1"] <- "845179"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZemaCp092.1"] <- "845187"
##assign mitochondrial accessions
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp008.1"] <- "4055864"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp024.1"] <- "4055860"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp032.1"] <- "4055888"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp033.1"] <- "4055938"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp036.1"] <- "4055894"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp050.1"] <- "4055956"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp051.1"] <- "4055881"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp057.1"] <- "4055973"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp081.1"] <- "4055890"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp092.1"] <- "4055837"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp097.1"] <- "4055857"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp098.1"] <- "4055896"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp113.1"] <- "4055802"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp119.1"] <- "4055928"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp129.1"] <- "4055878"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp136.1"] <- "4056004"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp159.1"] <- "4055794"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp167.1"] <- "4056007"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp179.1"] <- "4056006"
organelleEntrez$Entrez[organelleEntrez$Gene=="ZeamMp185.1"] <- "4055892"
##add organelle entrez to mgdb-assigned Entrez numbers 
Entrez <- mgdb[,c(1,12)]
colnames(Entrez) <- c("Gene", "Entrez")
Entrez <- Entrez[!Entrez$Entrez=="",]
Entrez <- rbind(Entrez, organelleEntrez)
Entrez <- unique(Entrez)
saveRDS(Entrez, "Entrez")

tmt4 <- join(x = tmt4, y = Entrez, by = "Gene", type="left", match="first")

tmt4 <- tmt4[,c(25,1:24)]
colnames(tmt4)
```
```{r remove pm and missing data}
n <- nrow(tmt4)
print(c("total proteins =", n), quote = F)
#remove non-maize proteins
tmt4 <- tmt4[(grep("Zm|Zema|Zeam", tmt4$Accession)),]
n2 <- nrow(tmt4)
print(c("non-maize proteins removed =", (n-n2)), quote = F)
#remove polymorphic proteins
tmt4.pm <- tmt4[grep("only", tmt4$Accession),]
tmt4 <- tmt4[-grep("only", tmt4$Accession),]
n3 <- nrow(tmt4)
print(c("polymorphic proteins removed =", (n2-n3)), quote = F)
#remove rows with missing tmt data
tmt4 <- tmt4[!tmt4[,10]==0,]
tmt4 <- tmt4[!tmt4[,11]==0,]
tmt4 <- tmt4[!tmt4[,12]==0,]
tmt4 <- tmt4[!tmt4[,13]==0,]
tmt4 <- tmt4[!tmt4[,14]==0,]
tmt4 <- tmt4[!tmt4[,15]==0,]
tmt4 <- tmt4[!tmt4[,16]==0,]
tmt4 <- tmt4[!tmt4[,17]==0,]
tmt4 <- tmt4[!tmt4[,18]==0,]
tmt4 <- tmt4[!tmt4[,19]==0,]
tmt4 <- tmt4[!tmt4[,20]==0,]
tmt4 <- tmt4[!tmt4[,21]==0,]
tmt4 <- tmt4[!tmt4[,22]==0,]
tmt4 <- tmt4[!tmt4[,23]==0,]
tmt4 <- tmt4[!tmt4[,24]==0,]
tmt4 <- tmt4[!tmt4[,25]==0,]
n4 <- nrow(tmt4)
print(c("proteins with missing data removed =", (n3-n4)), quote = F)
print(c("remaining proteins =", n4), quote = F)
```
```{r calculate ratios}
tmt4$MP.1 <- rowMeans(tmt4[,c("B.1","M.1")])
tmt4$MP.2 <- rowMeans(tmt4[,c("B.2","M.2")])
tmt4$MP.3 <- rowMeans(tmt4[,c("B.3","M.3")])
tmt4$MP.4 <- rowMeans(tmt4[,c("B.4","M.4")])

tmt4$l2.H1.MP1 <- log2(tmt4$H.1/tmt4$MP.1)
tmt4$l2.H2.MP2 <- log2(tmt4$H.2/tmt4$MP.2)
tmt4$l2.H3.MP3 <- log2(tmt4$H.3/tmt4$MP.3)
tmt4$l2.H4.MP4 <- log2(tmt4$H.4/tmt4$MP.4)

tmt4$l2.A1.B1 <- log2(tmt4$A.1/tmt4$B.1)
tmt4$l2.A2.B2 <- log2(tmt4$A.2/tmt4$B.2)
tmt4$l2.A3.B3 <- log2(tmt4$A.3/tmt4$B.3)
tmt4$l2.A4.B4 <- log2(tmt4$A.4/tmt4$B.4)
```
```{r BM/MP density}
tmt4.BM.MP <- tmt4[,c("Accession","Gene","l2.H1.MP1", "l2.H2.MP2", "l2.H3.MP3","l2.H4.MP4")]
colnames(tmt4.BM.MP) <- c("Accession","Gene","Section1", "Section2","Section3","Section4")
tmt4.BM.MP.m <- melt(tmt4.BM.MP)

tmt4.BM.MP.m$Category <- "Other"

tmt4.BM.MP.m$Category[tmt4.BM.MP.m$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs)] <- "PhANGs/PhAPGs"

tmt4.BM.MP.m$Category[tmt4.BM.MP.m$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo)] <- "Plastid ribosome"

tmt4.BM.MP.m$Category[tmt4.BM.MP.m$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"


colors.p2 <- c("Other"="grey50",  "Plastid ribosome"="#6A00EA",  "PhANGs/PhAPGs"="#008275", "Cytosolic ribosome"="#B50000")

tmt4.BM.MP.m$value[tmt4.BM.MP.m$value <= -2] <- -2
tmt4.BM.MP.m$value[tmt4.BM.MP.m$value >= 2] <- 2

ggplot(tmt4.BM.MP.m, mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p2, name="")+
  theme_ridges() +
  xlim(-2,2)+
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(BM/MP)")+
  guides(col=guide_legend(nrow = 2))+
  coord_cartesian(clip = "off", expand = F)
ggsave("tmt4.BM.MP.png", width = 4, height = 5)
```
```{r acs/B73 density}
tmt4.acs.B73 <- tmt4[,c("Accession","Gene","l2.A1.B1","l2.A2.B2", "l2.A3.B3","l2.A4.B4")]
colnames(tmt4.acs.B73) <- c("Accession","Gene","Section1", "Section2","Section3","Section4")
tmt4.acs.B73.m <- melt(tmt4.acs.B73)

tmt4.acs.B73.m$Category <- "Other"

tmt4.acs.B73.m$Category[tmt4.acs.B73.m$Gene %in% c(GOIs$PhANGs, GOIs$PhAPGs)] <- "PhANGs/PhAPGs"

tmt4.acs.B73.m$Category[tmt4.acs.B73.m$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo)] <- "Plastid ribosome"

tmt4.acs.B73.m$Category[tmt4.acs.B73.m$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"

colors.p2 <- c("Other"="grey50",  "Plastid ribosome"="#6A00EA",  "PhANGs/PhAPGs"="#008275", "Cytosolic ribosome"="#B50000")

tmt4.acs.B73.m$value[tmt4.acs.B73.m$value <= -2] <- -2
tmt4.acs.B73.m$value[tmt4.acs.B73.m$value >= 2] <- 2

ggplot(tmt4.acs.B73.m, mapping=aes(y=variable, x=value, color=Category))+
  geom_density_ridges(fill=NA, size=1, scale=1) +
  scale_color_manual(values = colors.p2, name="")+
  theme_ridges() +
  xlim(-2,2)+
  theme(legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), legend.text = element_text(size=9),)+
  geom_vline(xintercept = 0,  color="black", size=1)+
  xlab("log2(acs/B73)")+
  guides(col=guide_legend(nrow = 2))+
  coord_cartesian(clip = "off", expand = F)
ggsave("tmt4.acs.B73.png", width = 4, height = 5)
```


